File: src/main/java/model/Resource.java
package model;

public class Resource {
	public String name;
	public int offset;
	
	public Resource(String name, int offset) {
		this.name = name;
		this.offset = offset;
	}
	
	@Override
	public String toString() {
		return name+" "+offset;
	}
}

File: src/main/java/BTMaker/BTMaker/resources/Resource.kt
package BTMaker.BTMaker.resources

import java.io.DataInputStream
import java.io.DataOutputStream
import java.util.jar.JarFile

class Resource(var path: String, var offset: Int = 0, var length: Int? = null) {
    var type: ResourceType? = null

    fun getInputStream(jarFile: JarFile): DataInputStream {
        val entry = jarFile.getJarEntry(path)
            ?: throw IllegalArgumentException("Could not find resource file (/a)")
        return DataInputStream(jarFile.getInputStream(entry))
    }
}